### ISSUE TITLE
Stackoverflow while using annotation processing

### ISSUE THREAD

**Issue by shanon84** [michael-simons/neo4j-migrations#1116]
Hey there,
I am trying to use the neo4j-migrations-annotation-processor:2.3.1 in our project.
While compiling I get the following Exception (shortend, because it repeats itself):

```java.lang.StackOverflowError
	at jdk.compiler/com.sun.tools.javac.code.Symbol.apiComplete(Symbol.java:688)
	at jdk.compiler/com.sun.tools.javac.code.Symbol$ClassSymbol.getKind(Symbol.java:1476)
	at ac.simons.neo4j.migrations.annotations.proc.impl.CatalogGeneratingProcessor$RequiresPrimaryKeyConstraintPredicate.visitType(CatalogGeneratingProcessor.java:1066)
	at ac.simons.neo4j.migrations.annotations.proc.impl.CatalogGeneratingProcessor$RequiresPrimaryKeyConstraintPredicate.visitType(CatalogGeneratingProcessor.java:1041)
	at jdk.compiler/com.sun.tools.javac.code.Symbol$ClassSymbol.accept(Symbol.java:1573)
	at ac.simons.neo4j.migrations.annotations.proc.impl.CatalogGeneratingProcessor$RequiresPrimaryKeyConstraintPredicate.lambda$visitType$0(CatalogGeneratingProcessor.java:1071)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1856)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.noneMatch(ReferencePipeline.java:642)
	at ac.simons.neo4j.migrations.annotations.proc.impl.CatalogGeneratingProcessor$RequiresPrimaryKeyConstraintPredicate.visitType(CatalogGeneratingProcessor.java:1071)
	at ac.simons.neo4j.migrations.annotations.proc.impl.CatalogGeneratingProcessor$RequiresPrimaryKeyConstraintPredicate.visitType(CatalogGeneratingProcessor.java:1041)
	at jdk.compiler/com.sun.tools.javac.code.Symbol$ClassSymbol.accept(Symbol.java:1573)
	at ac.simons.neo4j.migrations.annotations.proc.impl.CatalogGeneratingProcessor$RequiresPrimaryKeyConstraintPredicate.lambda$visitType$1(CatalogGeneratingProcessor.java:1072)
	at java.base/java.util.Optional.map(Optional.java:260)
	at ac.simons.neo4j.migrations.annotations.proc.impl.CatalogGeneratingProcessor$RequiresPrimaryKeyConstraintPredicate.visitType(CatalogGeneratingProcessor.java:1072)
	at ac.simons.neo4j.migrations.annotations.proc.impl.CatalogGeneratingProcessor$RequiresPrimaryKeyConstraintPredicate.visitType(CatalogGeneratingProcessor.java:1041)
	at jdk.compiler/com.sun.tools.javac.code.Symbol$ClassSymbol.accept(Symbol.java:1573)
	at ac.simons.neo4j.migrations.annotations.proc.impl.CatalogGeneratingProcessor$RequiresPrimaryKeyConstraintPredicate.lambda$visitType$0(CatalogGeneratingProcessor.java:1071)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1856)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.noneMatch(ReferencePipeline.java:642)
	at ac.simons.neo4j.migrations.annotations.proc.impl.CatalogGeneratingProcessor$RequiresPrimaryKeyConstraintPredicate.visitType(CatalogGeneratingProcessor.java:1071)
	at ac.simons.neo4j.migrations.annotations.proc.impl.CatalogGeneratingProcessor$RequiresPrimaryKeyConstraintPredicate.visitType(CatalogGeneratingProcessor.java:1041)
	at jdk.compiler/com.sun.tools.javac.code.Symbol$ClassSymbol.accept(Symbol.java:1573)
	at ac.simons.neo4j.migrations.annotations.proc.impl.CatalogGeneratingProcessor$RequiresPrimaryKeyConstraintPredicate.lambda$visitType$1(CatalogGeneratingProcessor.java:1072)
	at java.base/java.util.Optional.map(Optional.java:260)
```

I asume it happens, because we are not using the correct neo4j annotations in our node classes for now.
It would be nice to get an understandable errormessage instead of a stackoverflow.

**Comment by michael-simons** [comment]
Hej, thanks for using this software and reporting the issue.

Would you be able to share some of the annotated classes? 
Are you using SDN6 or OGM annotations?

**Comment by michael-simons** [comment]
I assume you have quite the inheritance hierarchy in your domain class.
Can you share that plz?

**Comment by michael-simons** [comment]
Will be fixed in 2.7.1. Should work as a drop-in-replacement.

**Comment by michael-simons** [comment]
Thanks, and no worries.
I think I got it. It surely is a structure like this:

```java
import org.neo4j.ogm.annotation.Id;
import org.neo4j.ogm.annotation.NodeEntity;

/**
 * @author Michael J. Simons
 */
@NodeEntity
public abstract class Fun {

	@Id
	String uuid;

	static class FunFun extends Fun {
	}
}
```

(Doesn't matter whether OGM or SDN annotations)
